src/js/datadog-docs.js

```diff
@@ -503,30 +503,81 @@ $(document).ready(function () {
         }
     }
 
-    $('.code-tabs .tab-content').find('.tab-pane').each(function(idx, item) {
-      var navTabs = $(this).closest('.code-tabs').find('.nav-tabs'),
-          title = $(this).attr('title');
-      navTabs.append('<li><a href="#">'+title+'</a></li');
-    });
-
-    $('.code-tabs ul.nav-tabs').each(function() {
-      $(this).find("li:first").addClass('active');
-    })
+    if($('.code-tabs').length > 0) {
+        // page load set code tab titles
+        $('.code-tabs .tab-content').find('.tab-pane').each(function(idx, item) {
+          var navTabs = $(this).closest('.code-tabs').find('.nav-tabs'),
+              title = $(this).attr('title');
+          var lang = title.toLowerCase().replace(/\W/g, '');
+          navTabs.append('<li><a href="#" data-lang="'+lang+'">'+title+'</a></li');
+        });
 
-    $('.code-tabs .tab-content').each(function() {
-      $(this).find("div:first").addClass('active').addClass('show');
-    });
+        // page load if we have a lang in url activate those tabs, otherwise activate first
+        var sPageURL = decodeURIComponent(window.location.search.substring(1));
+        var sURLVariables = sPageURL.split('&');
+        var tab = sURLVariables.filter(function(item) {
+           return item.split('=')[0] === 'tab';
+        }).map(function(item) {
+            return item.split('=')[1];
+        }).toString();
+
+        function activateTab(el) {
+            var tab = el.parent(),
+                 tabIndex = tab.index(),
+                 tabPanel = el.closest('.code-tabs'),
+                 tabPane = tabPanel.find('.tab-pane').eq(tabIndex);
+             tabPanel.find('.active').removeClass('active');
+             tab.addClass('active');
+             tabPane.addClass('active');
+             tabPane.addClass('show');
+        }
 
-    $('.code-tabs .nav-tabs a').click(function(e){
-      e.preventDefault();
-      var tab = $(this).parent(),
-          tabIndex = tab.index(),
-          tabPanel = $(this).closest('.code-tabs'),
-          tabPane = tabPanel.find('.tab-pane').eq(tabIndex);
-      tabPanel.find('.active').removeClass('active');
-      tab.addClass('active');
-      tabPane.addClass('active');
-      tabPane.addClass('show');
-    });
+        // clicking a tab open them all
+        $('.code-tabs .nav-tabs a').click(function(e){
+          e.preventDefault();
+
+          // prepare
```

Comment:
@davidejones Thanks for commenting the code. Small thing (non-blocking), I find it easier to follow the comments if they are formatted as complete sentences. So, instead of `// prepare`, perhaps `// Calculate the top offset of the element so the value can be used to place it in front of the user after tab switch` (or whatever the actual calculation is for)